providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -1351,6 +1351,23 @@ static void *ec_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
     if (gctx->group_check != NULL)
         ret = ret && ossl_ec_set_check_group_type_from_name(ec,
                                                             gctx->group_check);
+
+#ifdef FIPS_MODULE
+    if (!ossl_fips_self_testing()
+        && EC_KEY_get0_public_key(ec) != NULL
+        && EC_KEY_get0_private_key(ec) != NULL
+        && EC_KEY_get0_group(ec) != NULL) {
```

Comment:
Again - shouldn't this only happen if we're actually generating a key pair? And if that is the case why would these other things ever be NULL?